### ISSUE TITLE
Circular dependency in (stepper.component <-> step.component) and (option.component<->select.component)

### ISSUE THREAD

**Issue by jagarapusrinivasarao** [akveo/nebular#742]
### Circular dependency

**I'm submitting an issue that I faced while running `npm start` command

* [ ] bug report

### Just run the  `npm start` and getting Circular dependency issue like
`
WARNING in Circular dependency detected:
src\framework\theme\components\select\option.component.ts -> src\framework\theme\components\select\select.component.ts -> src\framework\theme\components\select\option.component.ts

WARNING in Circular dependency detected:
src\framework\theme\components\select\select.component.ts -> src\framework\theme\components\select\option.component.ts -> src\framework\theme\components\select\select.component.ts

WARNING in Circular dependency detected:
src\framework\theme\components\stepper\step.component.ts -> src\framework\theme\components\stepper\stepper.component.ts -> src\framework\theme\components\stepper\step.component.ts

WARNING in Circular dependency detected:
src\framework\theme\components\stepper\stepper.component.ts -> src\framework\theme\components\stepper\step.component.ts -> src\framework\theme\components\stepper\stepper.component.ts`

**Current behavior:**
I can't even see the output in browser like the demo.
![image](https://user-images.githubusercontent.com/20349868/45948072-40922a80-c014-11e8-8f73-a6a923440ea2.png)

**Expected behavior:**
Should be compiled without errors.

**Steps to reproduce:**
>git clone https://github.com/akveo/ngx-admin.git
>cd ngx-admin
>npm install
>npm start

**Related code:**

```
insert short code snippets here
```

### Other information:

**npm, node, OS, Browser**
```
npm --version
5.5.1

npm --version
5.5.1

OS: Windows 10
Browser: Chrome/Firefox
```

**Angular, Nebular**
```

Angular CLI: 6.0.3
Nebular : 2.0.0-rc.10
```

**Comment by nnixaa** [comment]
Hi @jagarapusrinivasarao, I'm pretty sure there is some mistake in the steps to reproduce:

```
git clone akveo/ngx-admin.git
cd ngx-admin
npm install
npm start
```
as the output you received is for Nebular repo, not ngx-admin.
Still leaving this issue as we need to get rid of those warnings.

**Comment by Hyperonic** [comment]
Hi @nnixaa, I've cloned it for learning purpose as Im indirectly using it through ngx-admin

**Comment by nnixaa** [comment]
fixed in 4.0
